src/Streamly/String.hs

```diff
@@ -177,18 +194,56 @@ isSpace c
   where
     uc = fromIntegral (ord c) :: Word
 
+-- | Break a string up into a list of strings at newline characters.
+-- The resulting strings do not contain newlines.
+--
+-- >>> S.toList $ lines $ S.fromList "lines\nthis\nstring\n\n\n"
+-- ["lines","this","string","",""]
+--
+-- If you're dealing with lines of massive length, consider using
+-- 'foldLines' instead.
 {-# INLINE lines #-}
 lines :: (MonadIO m, IsStream t) => t m Char -> t m (Array Char)
 lines = FL.splitSuffixBy (== '\n') toArray
 
+-- | Break a string up into a list of strings, which were delimited
+-- by characters representing white space.
+--
+-- >>> S.toList $ words $ S.fromList "A  newline\nis considered white space?"
+-- ["A", "newline", "is", "considered", "white", "space?"]
+--
+-- If you're dealing with words of massive length, consider using
+-- 'foldWords' instead.
 {-# INLINE words #-}
 words :: (MonadIO m, IsStream t) => t m Char -> t m (Array Char)
 words = FL.wordsBy isSpace toArray
 
+-- | Flattens the stream of @Array Char@, after appending a terminating
+-- newline to each string.
+--
+-- 'unlines' is an inverse operation to 'lines'.
+--
+-- >>> S.toList $ unlines $ S.fromList ["lines", "this", "string"]
```

Comment:
We may need to mention `OverloadedStrings` extension wherever we are relying on it. Novice users may not be able to figure that out easily and they may assume that the example just fails.